src/EditorFeatures/CSharpTest/Intents/DeleteParameterIntentTests.cs

```diff
@@ -0,0 +1,527 @@
+    }
+}
+
+public static class FooExtensions
+{
+    public static void DoFoo(this {|priorSelection:Foo|} foo)
+    {
+
+    }
+}";
+        var currentText =
+@"public class Foo
+{
+    static void Bar()
+    {
+        var f = new Foo();
+        f.DoFoo();
+    }
+}
+
+public static class FooExtensions
+{
+    public static void DoFoo()
+    {
+
+    }
+}";
+
+        await VerifyIntentMissingAsync(WellKnownIntents.DeleteParameter, initialText, currentText).ConfigureAwait(false);
+    }
+
+    [Fact]
+    public async Task TestDeleteParameterInExtensionMethodAsync()
+    {
+        var initialText = @"public class Foo
+{
+    static void Bar()
+    {
+        var f = new Foo();
+        f.DoFoo({|priorSelection:1|}, 2);
+    }
+}
+
+public static class FooExtensions
+{
+    public static void DoFoo(this Foo foo, int value1, int value2)
+    {
+
+    }
+}";
+        var currentText =
+@"public class Foo
+{
+    static void Bar()
+    {
+        var f = new Foo();
+        f.DoFoo(2);
+    }
+}
+
+public static class FooExtensions
+{
+    public static void DoFoo(this Foo foo, int value1, int value2)
+    {
+
+    }
+}";
+        var expectedText =
+@"public class Foo
+{
+    static void Bar()
+    {
+        var f = new Foo();
+        f.DoFoo(2);
+    }
+}
+
+public static class FooExtensions
+{
+    public static void DoFoo(this Foo foo, int value2)
+    {
+
+    }
+}";
+
+        await VerifyExpectedTextAsync(WellKnownIntents.DeleteParameter, initialText, currentText, expectedText).ConfigureAwait(false);
+    }
+
+    [Fact]
+    public async Task TestDeleteParameterOnDefinitionAsync()
+    {
+        var initialText = @"public class Foo
+{
+    static void Bar()
+    {
+        var f = new Foo();
+        f.DoFoo(1, 2);
+    }
+}
+
+public static class FooExtensions
+{
+    public static void DoFoo(this Foo foo, int {|priorSelection:value1|}, int value2)
+    {
+
+    }
+}";
+        var currentText =
+@"public class Foo
+{
+    static void Bar()
+    {
+        var f = new Foo();
+        f.DoFoo(2);
+    }
+}
+
+public static class FooExtensions
+{
+    public static void DoFoo(this Foo foo, int value1, int value2)
+    {
+
+    }
+}";
+        var expectedText =
+@"public class Foo
+{
+    static void Bar()
+    {
+        var f = new Foo();
+        f.DoFoo(2);
+    }
+}
+
+public static class FooExtensions
+{
+    public static void DoFoo(this Foo foo, int value2)
+    {
+
+    }
+}";
+
+        await VerifyExpectedTextAsync(WellKnownIntents.DeleteParameter, initialText, currentText, expectedText).ConfigureAwait(false);
+    }
+
+    [Fact]
+    public async Task TestDeleteParamsParameterAsync()
+    {
+        var initialText =
+@"class C
+{
+    void M()
+    {
+        Method(new C(), {|priorSelection:1|}, 2, 3);
+    }
+
+    void Method(C c, params int[] values)
+    {
+    }
+}";
+        var currentText =
+@"class C
+{
+    void M()
+    {
+        Method(new C(), 2, 3);
+    }
+
+    void Method(C c, params int[] values)
```

Comment:
I'm surprised to see we are offering to delete params parameter when user delete only one of the arguments that feed into the params. Is this the intended behavior?

Reply:
I am slightly leaning towards offering it just because intellicode has to be fairly confident a parameter deletion is happening (e.g. you have deleted the same parameter in multiple instances).  THey should also be sending us the entire parameter span.  It might be worth blocking this on a single params param, but today the change signature code allows this to happen.  

After taking a quick look, it could be slightly complex to block - so for now I'll leave it as-is until I'm able to try it out with the intellicode integration.  If it turns out they're detecting it incorrectly and sending the partial param span, then we can look at actually blocking it.

Reply:
SGTM!